internal/rbd/rbd_attach.go

```diff
@@ -207,21 +225,36 @@ func setRbdNbdToolFeatures() {
 		_, stderr, err = util.ExecCommand(context.TODO(), "modprobe", moduleNbd)
 		if err != nil {
 			hasNBD = false
-			log.WarningLogMsg("rbd-nbd: nbd modprobe failed (%v): %q", err, stderr)
+			log.WarningLogMsg("nbd modprobe failed (%v): %q", err, stderr)
 		}
 	}
-
-	stdout, stderr, err := util.ExecCommand(context.TODO(), rbdTonbd, "--help")
-	if err != nil || stderr != "" {
-		hasNBD = false
-		log.WarningLogMsg("running rbd-nbd --help failed with error:%v, stderr:%s", err, stderr)
+	if hasNBD {
```

Comment:
return if the nbd module is not found. we don't need to check cookie support if nbd itself is not present.

Reply:
yes, I agree, in general we take the approach of error-checking/validation, and immediately returning if something is not right, only continue if all is OK.